File: 3702. Longest Subsequence With Non-Zero Bitwise XOR.py
from typing import List
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        xr=0
        count=0

        for num in nums:
            xr^=num
            if(num!=0):
                count+=1

        if(xr!=0):
            return len(nums)
        elif(count>0):
            return len(nums)-1
        
        return 0

File: 3702. Longest Subsequence With Non-Zero Bitwise XOR.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int count=0;
        int xr=0;

        for(int num : nums){
            xr^=num;
            if(num!=0)
                count=num;
        }
        if(xr!=0)
            return nums.size();
        else
            if(count!=0)
                return nums.size()-1;
        
        return 0;
    }
};